server/routes/conversation.js

```diff
@@ -0,0 +1,17 @@
+const express = require("express");
+const router = express.Router();
+const protect = require("../middleware/auth");
+const multer = require("multer");
+
+const {
+  getConversations,
+  createConversation
+} = require("../controllers/conversation");
+
+
+router.route("/").get(protect, getConversations);
+
+router.route("/").post(protect, createConversation);
```

Comment:
> Though I'm not sure if there's a need for `createConversation` route, a conversation should be created in the `createMessage` controller. Anytime a new message is sent, you first check if there's an existing conversation between the 2 users. If there is, go ahead and create the message using that `convoId`, otherwise, create a new conversation before creating the message with the new `convoId`. This is my idea of how it works.

You are right, I miss the logic.